### PR TITLE
Test for updating repositories with the same dependency

### PR DESCRIPTION
**Pull request by MilenaHristova**
https://github.com/dotnet/arcade/issues/10798

### CODE REVIEW

File: src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.VirtualMonoRepo.E2E.Tests/VmrPatchesTestsBase.cs [hunk 1]
```diff
@@ -16,6 +16,7 @@ public class VmrPatchesTestsBase : VmrTestsBase
 {
     protected string patchFileName = null!;
     protected LocalPath installerPatchesDir = null!;
+    protected LocalPath installerFilePath = null!;
```
Comment (premun) [inline src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.VirtualMonoRepo.E2E.Tests/VmrPatchesTestsBase.cs:19]: I noticed this already when writing my tests - protected members should be properties and capitalized.

So ideally:
```csharp
protected LocalPath InstallerFilePath { get; private set; } = null!;
```

File: src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.VirtualMonoRepo.E2E.Tests/VmrRecursiveSyncTests.cs [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+            {
+                Constants.InstallerRepoName,
+                new List<string>
+                {
+                    Constants.ProductRepoName, 
+                    Constants.SubmoduleRepoName
+                }
+            },
+            {Constants.ProductRepoName, new List<string> {Constants.DependencyRepoName} },
+            {Constants.SubmoduleRepoName, new List<string> {Constants.DependencyRepoName }},
```
Comment (premun) [inline src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.VirtualMonoRepo.E2E.Tests/VmrRecursiveSyncTests.cs:93]:
```suggestion
            { Constants.ProductRepoName, new List<string> { Constants.DependencyRepoName } },
            { Constants.SubmoduleRepoName, new List<string> { Constants.DependencyRepoName }},
```

File: src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.VirtualMonoRepo.E2E.Tests/VmrRecursiveSyncTests.cs [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+            expectedFilesFromRepos);
+
+        CheckDirectoryContents(_vmrPath, expectedFiles);
+
+        // create new version of dependency repo
+
+        File.WriteAllText(_dependencyRepoPath / Constants.DependencyRepoFileName, "New version of the file");
+        await GitOperations.CommitAll(_dependencyRepoPath, "change the file in dependency repo");
+
+        // the second repo depends on the new version, first repo depends on the old one
```
Comment (premun) [inline src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.VirtualMonoRepo.E2E.Tests/VmrRecursiveSyncTests.cs:50]: Would it be possible to have somewhere a comment with the dependency tree? Similarly as to the original issue?

File: src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.VirtualMonoRepo.E2E.Tests/VmrSyncRepoChangesTest.cs [hunk 4]
```diff
@@ -158,9 +157,9 @@ public async Task SubmodulesAreInlinedProperlyTest()
 
     protected override async Task CopyReposForCurrentTest()
     {
-        var dependenciesMap = new Dictionary<string, List<Dependency>>
+        var dependenciesMap = new Dictionary<string, List<string>>
         {
-            {Constants.ProductRepoName,  new List<Dependency> {new Dependency(Constants.DependencyRepoName, _dependencyRepoPath) } }
+            {Constants.ProductRepoName,  new List<string> {Constants.DependencyRepoName} }
```
Comment (premun) [inline src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.VirtualMonoRepo.E2E.Tests/VmrSyncRepoChangesTest.cs:162]:
```suggestion
            { Constants.ProductRepoName,  new List<string> { Constants.DependencyRepoName } }
```

File: src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.VirtualMonoRepo.E2E.Tests/VmrRecursiveSyncTests.cs (hunk 3 above)
Comment (MilenaHristova) [reply src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.VirtualMonoRepo.E2E.Tests/VmrRecursiveSyncTests.cs:50]: yes, good idea

**Review by premun — APPROVED** [review]
This reads much better now, thanks!

**Pull request merged**